src/main/java/org/broadinstitute/hellbender/tools/IndexFeatureFile.java

```diff
@@ -104,13 +105,20 @@ private File determineFileName(final Index index) {
     }
 
     private Index createAppropriateIndexInMemory(final FeatureCodec<? extends Feature, ?> codec ) {
-        // For block-compression VCF files, write a Tabix index
+        // For block-compression files, write a Tabix index
         if ( AbstractFeatureReader.hasBlockCompressedExtension(featureFile) ) {
-            if ( isVCFCodec(codec) ) {
-                return IndexFactory.createTabixIndex(featureFile, codec, TabixFormat.VCF, null);
-            }
-            else {
-                throw new UserException("This tool only supports indexing of block-compressed files when they are in VCF format");
+            try {
+                // TODO: this could benefit from provided sequence dictionary from reference
+                // TODO: this can be an optional parameter for the tool
+                return IndexFactory
+                        .createIndex(featureFile, codec, IndexFactory.IndexType.TABIX, null);
```

Comment:
Do you have any suggestion for the TODO? I think that an optional parameter with the sequence dictionary will be interesting for fast indexing...